### PR TITLE
Fix not matchers

### PR DESCRIPTION
**Pull request by mgrybyk**
* revert changes in `utils.ts` related to `waitUntil` function logic introduced in https://github.com/webdriverio/expect-webdriverio/pull/133
* add tests for `not` matchers
* also tested changes in my real project

fixes: https://github.com/webdriverio/expect-webdriverio/issues/168

### CODE REVIEW

**Comment by mgrybyk** [comment]
let's increase coverage with https://github.com/webdriverio/expect-webdriverio/issues/169

File: test/matchers/mock/toBeCalled.test.ts [hunk 1]
```diff
@@ -47,13 +47,13 @@ describe('toBeRequested', () => {
 
         // expect(mock).not.toBeRequested() should pass
         const result = await toBeRequested.call({ isNot: true }, mock)
-        expect(result.pass).toBe(true)
+        expect(result.pass).toBe(false)
```
Comment (christian-bromann) [inline test/matchers/mock/toBeCalled.test.ts:50]: Shouldn't `expect(mock).not.toBeRequested()` pass here?

File: test/matchers/mock/toBeCalledTimes.test.ts [hunk 2]
```diff
@@ -79,21 +80,21 @@ describe('toBeRequestedTimes', () => {
 
         // expect(mock).not.toBeRequestedTimes(0) should fail
         const result = await toBeRequestedTimes.call({ isNot: true }, mock, 0)
-        expect(result.pass).toBe(false)
+        expect(result.pass).toBe(true)
```
Comment (christian-bromann) [inline test/matchers/mock/toBeCalledTimes.test.ts:83]: same here .. `expect(mock).not.toBeRequestedTimes(0)` should mean the mock was requested at least 1 time or more to pass but it was not requested so far which should make the assertion `expect(result.pass).toBe(false)` pass.

**Review by christian-bromann — CHANGES_REQUESTED** [review]
I feel like some of the assertions should stay as is or I am confusing something.

**Comment by mgrybyk** [comment]
This is just the same confusion I've faced.

If isNot is true and result.pass is true - the assertion would fail.
In order assertion to succeed when this.isNot is true result.pass should be false.

I spent some time going crazy with all that, that's why Ive added so many tests

**Comment by mgrybyk** [comment]
Please see how other matchers work, ex test/matchers/beMatchers.test.ts

**Review by christian-bromann — APPROVED** [review]
I see, yeah that is pretty confusing 😆

**Comment by mgrybyk** [comment]
resolved the conflicts

**Pull request merged**